### PR TITLE
net-irc/kvirc: add 5.2.6

### PR DESCRIPTION
**Pull request by DarthGandalf**
---

Please check all the boxes that apply:

- [ ] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [ ] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [ ] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [ ] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: net-irc/kvirc/kvirc-5.2.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		-DWANT_PERL=$(usex perl)
+		-DWANT_PHONON=OFF
+		-DWANT_PYTHON=$(usex python)
+		-DWANT_QTDBUS=$(usex dbus)
+		-DWANT_QTWEBENGINE=$(usex webengine)
+		-DWANT_SPELLCHECKER=$(usex spell)
+		-DQT_VERSION_MAJOR=6
+
+		# COMPILE_SVG_SUPPORT not used in source code.
+		-DWANT_QTSVG=OFF
```
Comment (a17r) [inline net-irc/kvirc/kvirc-5.2.6.ebuild:118]:
```suggestion
		-DWANT_DEBUG=$(usex debug)
		-DWANT_DOXYGEN=$(usex doc)
		-DWANT_GETTEXT=$(usex nls)
		-DWANT_GSM=$(usex gsm)
		-DWANT_KDE=$(usex kde)
		-DWANT_MEMORY_PROFILE=$(usex profile)
		-DWANT_OGG_THEORA=$(usex theora)
		-DWANT_OPENSSL=$(usex ssl)
		-DWANT_OSS=$(usex oss)
		-DWANT_PERL=$(usex perl)
		-DWANT_PYTHON=$(usex python)
		-DWANT_QTDBUS=$(usex dbus)
		-DWANT_QTWEBENGINE=$(usex webengine)
		-DWANT_SPELLCHECKER=$(usex spell)
		-DQT_VERSION_MAJOR=6

		-DWANT_DCC_VIDEO=OFF
		-DWANT_PHONON=OFF
		-DWANT_QTSVG=OFF # COMPILE_SVG_SUPPORT not used in source code.
```
(just preempting existing ebuild style)

File: net-irc/kvirc/kvirc-5.2.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+src_install() {
+	cmake_src_install
+
+	if use doc; then
+		(
+			docinto html
+			dodoc -r "${BUILD_DIR}/doc/api/html/"*
+		)
```
Comment (a17r) [inline net-irc/kvirc/kvirc-5.2.6.ebuild:151]: Why the brackets?

File: net-irc/kvirc/kvirc-5.2.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		# COMPILE_SVG_SUPPORT not used in source code.
+		-DWANT_QTSVG=OFF
+	)
+	if use python; then
+		mycmakeargs+=(
+			-DPython3_INCLUDE_DIR="$(python_get_includedir)"
+			-DPython3_LIBRARY="$(python_get_library_path)"
+		)
+	fi
+	if [[ "${PV}" == "9999" ]]; then
```
Comment (a17r) [inline net-irc/kvirc/kvirc-5.2.6.ebuild:126]:
```suggestion
	if [[ ${PV} == *9999* ]]; then
```

File: net-irc/kvirc/kvirc-5.2.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+DESCRIPTION="Advanced IRC Client"
+HOMEPAGE="https://www.kvirc.net/ https://github.com/kvirc/KVIrc"
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit cmake flag-o-matic python-single-r1 xdg
+
+if [[ "${PV}" == "9999" ]]; then
```
Comment (a17r) [inline net-irc/kvirc/kvirc-5.2.6.ebuild:11]:
```suggestion
if [[ ${PV} == *9999* ]]; then
```

File: net-irc/kvirc/kvirc-5.2.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/kvirc/KVIrc"
+else
+	SRC_URI="https://github.com/kvirc/KVIrc/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~ppc ~ppc64 ~riscv ~x86"
+	S="${WORKDIR}/KVIrc-${PV}"
```
Comment (a17r) [inline net-irc/kvirc/kvirc-5.2.6.ebuild:17]:
```suggestion
	SRC_URI="https://github.com/kvirc/KVIrc/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
	S="${WORKDIR}/KVIrc-${PV}"
	KEYWORDS="~amd64 ~ppc ~ppc64 ~riscv ~x86"
```

**Review by a17r — CHANGES_REQUESTED** [review]
Maybe in addition you could look into python3_13 compatibility?

File: net-irc/kvirc/kvirc-5.2.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+REQUIRED_USE="audiofile? ( oss ) python? ( ${PYTHON_REQUIRED_USE} )"
+
+BDEPEND="
+	dev-lang/perl:0
+	virtual/pkgconfig
+	doc? ( ( || <app-text/doxygen-1.11.0 >=app-text/doxygen-1.12.0 ) )
+	kde? ( kde-frameworks/extra-cmake-modules:0 )
+	nls? ( sys-devel/gettext )"
+DEPEND="
+	dev-qt/qtbase:6[concurrent,gui,network,sql,widgets,xml]
```
[the remark this answers is not in the document]
Comment (DarthGandalf) [reply net-irc/kvirc/kvirc-5.2.6.ebuild:33]: It doesn't. That's why it depends on libX11 directly, and not via Qt :)

File: net-irc/kvirc/kvirc-5.2.6.ebuild [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+	kde? (
+		kde-frameworks/kconfig:6
+		kde-frameworks/kcoreaddons:6
+		kde-frameworks/kio:6
+		kde-frameworks/ki18n:6
+		kde-frameworks/knotifications:6
+		kde-frameworks/kparts:6
+		kde-frameworks/kservice:6
+		kde-frameworks/kstatusnotifieritem:6
+		kde-frameworks/kwindowsystem:6
```
[the remark this answers is not in the document]
Comment (DarthGandalf) [reply net-irc/kvirc/kvirc-5.2.6.ebuild:48]: You're right, fixed.

File: net-irc/kvirc/kvirc-5.2.6.ebuild (hunk 2 above)
Comment (DarthGandalf) [reply net-irc/kvirc/kvirc-5.2.6.ebuild:151]: Removed.

After inheriting this ebuild from previous maintainer I didn't try fix all the preexisting issues

**Review by a17r — APPROVED** [review]

**Pull request closed without merging**